107/src/main/java/org/ehcache/jsr107/Eh107CacheManager.java

```diff
@@ -219,6 +224,22 @@ public Properties getProperties() {
       cacheLoaderWriterFactory.registerJsr107Loader(cacheName, cacheLoaderWriter);
     }
 
+    //If cache is not read or write thro' remove the loader writer configuration from the JSR 107 conf
+    if(!jsr107Config.isReadThrough() && !jsr107Config.isWriteThrough()) {
+      DefaultCacheLoaderWriterConfiguration conf = builder.getExistingServiceConfiguration(DefaultCacheLoaderWriterConfiguration.class);
+      if(conf != null) {
+        LOG.warn("Removing the loader/writer service configuration from the JSR107 cache {}", cacheName);
+        builder.removeServiceConfig(conf);
+      }
+    }
+    if((jsr107Config.isReadThrough() || jsr107Config.isWriteThrough()) && cacheLoaderWriter == null) {
+      DefaultCacheLoaderWriterConfiguration conf = builder.getExistingServiceConfiguration(DefaultCacheLoaderWriterConfiguration.class);
+      if(conf == null) {
+        throw new InstantiationException("Unable to construct (read/write)through cache without either a templated loader/writer or configured loader/writer");
+      }
+      LOG.warn("Using the templated cache loader/writer {} for the cache {}", conf.getClazz().getName(), cacheName);
+    }
+
```

Comment:
(general comment on the whole block above) I don't like that we end up with 3 independent if blocks (1 from before + 2 added here). Can this be refactored in if/else rather?
Also I believe we should log the case where a config ends up overriding a template provided `CacheLoaderWriter`.